src/matching/network.ts

```diff
@@ -19,6 +19,28 @@ function getUrlAfterHostname(
   return url.substring(url.indexOf(hostname) + hostname.length);
 }
 
+// pattern$fuzzy
+function checkPatternFuzzyFilter(filter: NetworkFilter, request) {
+  const signature = filter.getFuzzySignature();
+
+  // Only compute this once, lazily
+  if (request.fuzzySignature === undefined) {
+    request.fuzzySignature = createFuzzySignature(request.url);
+  }
+
+  const requestSignature = request.fuzzySignature;
+  let lastIndex = 0;
+  for (let i = 0; i < signature.length; i += 1) {
```

Comment:
maybe we can also check the length to exit earlier

Reply:
I'm not sure I understood which length you suggest to check.

Reply:
I meant something like `if (signature.length > requestSignature.length) { return false; }`

Reply:
Yes true. Not sure how common it will be to have urls with less tokens than filters, but I will add it! Thanks for the suggestion.

Reply:
True, probably happens only for very short urls